Assets/Source/Components/UI/HealthBarComponent.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Assets.Source.Components.Actor;
+using Assets.Source.Components.UI.Base;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Assets.Source.Components.UI
+{
+    public class HealthBarComponent : HUDBase
+    {
+        private readonly float LERP_THRESHOLD = .025f;
+
+        private Animator Animator;
+        private ActorBehavior ActorBehavior;
+        private Image healthImage;
+        private int presentedHealth;
+
+        public override void ComponentStart()
+        {
+            Animator = GetRequiredComponent<Animator>();
+            healthImage = GetRequiredComponent<Image>();
+            ActorBehavior = GetPlayerActorBehavior();
+
+            presentedHealth = ActorBehavior.Health;
+
+            base.ComponentStart();
+        }
+
+        public override void ComponentUpdate()
+        {
+            if(ActorBehavior.Health != presentedHealth)
+            {
+                RunHealthAnimation();
+            }
+            else
+            {
+                StopHealthAnimation();
+            }
+            base.ComponentUpdate();
+        }
+
+        public void StopHealthAnimation()
+        {
+            Animator.SetBool("highlight", false);
+        }
+
+        public void RunHealthAnimation()
+        {
+            Animator.SetBool("highlight", true);
+
+            float startingPercent = (float)presentedHealth / ActorBehavior.MaxHealth;
```

Comment:
this is fine i guess, but i still think you're overthinking this logic lol.

This could literally be:

```
int presentedHealth = actor.health;

update() {
    if (actor.health < presentedHealth) {
        animator.SetBool("highlihglt", true);
        presentedHealth--;
    } else {
        animator.SetBOol("higlight", false);
    }

   healthImage.fillAmount = (presentedHealth / actor.MaxHealth);
}
```
That'd make the health bar just follow whatever the actual health is at a slower rate

Reply:
You're right fam it can be simpler, i was trying to do something fancy by making the health bar rapidly decrease then slowly decrease as it got closer to the actual value of the health, but i can keep it simple for now